### PR TITLE
User termination check in MIP solver

### PR DESCRIPTION
**Pull request by andreaslundell**
Ok, I think I have a solution to https://github.com/coin-or/SHOT/issues/86, but could you verify @svigerske that it actually works from GAMS (with Cbc, Cplex and Gurobi).

If it works, I will still perform some more extensive benchmarks to make sure this fix does not have any unforseen consequences.

### CODE REVIEW

**Comment by andreaslundell** [comment]
I would suggest you try this fix on the SHOT_NLP branch if you want to reformulate the `tln*` instances. There were some bug fixes there that makes the reformulations work better.

**Comment by svigerske** [comment]
With CPLEX, yes. Gurobi is difficult to test, but it should work, too :).
For Cbc, you didn't finish the implementation, I believe.

**Comment by andreaslundell** [comment]
Sorry about that. The Cbc version should be working now.

File: src/MIPSolver/MIPSolverCbc.cpp [hunk 1]
```diff
@@ -31,6 +32,44 @@
...
+
+    virtual TerminationEventHandler* clone() const { return new TerminationEventHandler(env); };
+
+    virtual CbcAction event(CbcEvent whichEvent)
+    {
+        if(whichEvent == CbcEventHandler::CbcEvent::node)
+        {
+            env->output->outputDebug("        Terminated by user.");
+
+            // return (CbcEventHandler::CbcAction::stop);
```
Comment (svigerske) [inline src/MIPSolver/MIPSolverCbc.cpp:63]: Shouldn't it check with SHOT here whether it should return `stop` ?

Comment (andreaslundell) [reply src/MIPSolver/MIPSolverCbc.cpp]: > Shouldn't it check with SHOT here whether it should return `stop` ?

Yes, it should :)

**Pull request merged**